### ISSUE TITLE
Failure to set DATA_FILE_PRINT_FORMAT to new value

### ISSUE THREAD

**Issue by ArtPoon** [veg/hyphy#329]
At commit fb49307ea76697f0f1089ff0d30190e73cecc132, binary at version `HYPHY 2.220150916beta(MP) for Darwin on x86_64`.  Using HyPhy as a Python shared library (version string was extracted through Python by calling GetString).  

Running the following test script:
```
import HyPhy
h = HyPhy._THyPhy('', 1)
res = h.ExecuteBF('return DATA_FILE_PRINT_FORMAT;', False)
print res.sData
h.ExecuteBF('DATA_FILE_PRINT_FORMAT=0;', False)
res = h.ExecuteBF('return DATA_FILE_PRINT_FORMAT;', False)
print res.sData
```

yields the following output:
```
6
6
```
instead of the expected output:
```
6
0
```

**Comment by spond** [comment]
Hi Art,

Hmm, this is not good. I'll check what's going on...

Sergei

**Comment by ArtPoon** [comment]
The problem seems to be at `site.cpp` line 5768:
```
checkParameter (dataFilePrintFormat,dFPrintFormat,6.0);
```
The value `dataFilePrintFormat` is set in `likefunc.cpp` -- it might not be getting carried over so this statement defaults to `6`.

**Comment by spond** [comment]
Hi Art,

Can you try to see what happens if you set the value of **any** variable and see if it sticks between calls? (e.g. "x")

`checkParameter` does not modify HBL variable values (only internal C++ variable values), so that's not the problem. I am thinking there may be a lack of state persistence between calls to `ExecuteBF`, so everything gets reset to default values.

Sergei

**Comment by ArtPoon** [comment]
This script:
```
import HyPhy
h = HyPhy._THyPhy('', 1)
res = h.ExecuteBF('GetString(v, HYPHY_VERSION, 1);', False)
res = h.ExecuteBF('return v;', False)
print res.sData

res = h.ExecuteBF('return ACCEPT_ROOTED_TREES;', False)
print 'before', res.sData
h.ExecuteBF('ACCEPT_ROOTED_TREES=1;', False)
res = h.ExecuteBF('return ACCEPT_ROOTED_TREES;', False)
print 'after', res.sData
```

returns the following:
```
HYPHY 2.220150916beta(MP) for Darwin on x86_64
before 0
after 1
```
However, if I replace `ACCEPT_ROOTED_TREES` with `LIKELIHOOD_FUNCTION_OUTPUT`, then I get:
```
HYPHY 2.220150916beta(MP) for Darwin on x86_64
before 2
after 2
```
So this persistence issue seems to affect only a subset of environmental variables.

**Comment by spond** [comment]
Hi Art, 

There was an unconditional call to `applyPreferences`, which would override the values of some globals when you called `ExecuteBF`. This is fixed in the v2.2.6 branch with eacd0492923fd5d07c1aec3dd4c17727844521cb 

```
import HyPhy


def setNecho (name, value, instance):
    res = instance.ExecuteBF('return %s;' % name, False)
    print ('before', res.sData)
    res = instance.ExecuteBF('%s=%g; return %s;' % (name, value, name), False)
    print ('during', res.sData)
    res = instance.ExecuteBF('return %s;' % name, False)
    print ('after', res.sData)

setNecho ("LIKELIHOOD_FUNCTION_OUTPUT", 5., HyPhy._THyPhy('', 1))
```

returns 

```
before 0
during 5
after 5
```

Sergei

**Comment by ArtPoon** [comment]
Fixed, thanks.
